dbms/src/Functions/FunctionsString.cpp

```diff
@@ -4381,6 +4381,358 @@ class FunctionRepeat : public IFunction
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        auto first_argument = arguments[0];
+        if (!first_argument->isString())
+            throw Exception(
+                fmt::format("Illegal type {} of first argument of function {}", arguments[0]->getName(), getName()),
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        auto second_argument = arguments[1];
+        if (!second_argument->isString())
+            throw Exception(
+                fmt::format("Illegal type {} of first argument of function {}", arguments[1]->getName(), getName()),
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        return std::make_shared<DataTypeInt64>();
+    }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result) const override
+    {
+        auto c0_col_column = block.getByPosition(arguments[0]).column;
+        auto c1_col_column = block.getByPosition(arguments[1]).column;
+        size_t val_num = block.rows();
+
+        auto col_res = ColumnInt64::create();
+        col_res->reserve(val_num);
+        ColumnInt64::Container & res_vec = col_res->getData();
+        res_vec.resize(val_num);
+
+        if (const auto * str_const_string = checkAndGetColumnConst<ColumnString>(c0_col_column.get()))
+        {
+            const auto * substr_const_string = checkAndGetColumnConst<ColumnString>(c1_col_column.get());
+            if (!substr_const_string)
+            {
+                const auto * substr_vector_string = checkAndGetColumn<ColumnString>(c1_col_column.get());
+                const auto * str_const_string_value = checkAndGetColumn<ColumnString>(str_const_string->getDataColumnPtr().get());
+                constVector(str_const_string_value->getChars(), str_const_string_value->getOffsets(), substr_vector_string->getChars(), substr_vector_string->getOffsets(), val_num, res_vec);
+            }
+        }
+        else
+        {
+            const auto * str_vector_string = checkAndGetColumn<ColumnString>(c0_col_column.get());
+            if (const auto * substr_const_string = checkAndGetColumnConst<ColumnString>(c1_col_column.get()))
+            {
+                const auto * substr_const_string_value = checkAndGetColumn<ColumnString>(substr_const_string->getDataColumnPtr().get());
+                vectorConst(str_vector_string->getChars(), str_vector_string->getOffsets(), substr_const_string_value->getChars(), substr_const_string_value->getOffsets(), val_num, res_vec);
+            }
+            else
+            {
+                const auto * substr_vector_string = checkAndGetColumn<ColumnString>(c1_col_column.get());
+                vectorVector(str_vector_string->getChars(), str_vector_string->getOffsets(), substr_vector_string->getChars(), substr_vector_string->getOffsets(), val_num, res_vec);
+            }
+        }
+
+        block.getByPosition(result).column = std::move(col_res);
+    }
+
+private:
+    static Int64 find(
```

Comment:
It looks like we can use Volnitsky directly.
`#include <Common/Volnitsky.h>`
Ref https://github.com/pingcap/tiflash/blob/10ebe6bf6866bae2a8f213e1b405d4aa1cd5781f/dbms/src/Functions/FunctionsString.cpp#L4635